src/main/java/eu/europeana/metis/sandbox/domain/Event.java

```diff
@@ -0,0 +1,59 @@
+package eu.europeana.metis.sandbox.domain;
+
+import static java.util.Objects.requireNonNull;
+
+import eu.europeana.metis.sandbox.common.Status;
+import eu.europeana.metis.sandbox.common.Step;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+public class Event<T extends Record> {
```

Comment:
Is generic parameter T ever anything other than equal to Record? Do we need to apply generics here?

Reply:
Not really, I was experimenting, but you are right is not going to be anything else than a Record, I will change it